src/tools/idlc/src/descriptor.c

```diff
@@ -1548,42 +1553,53 @@ static int print_opcode(FILE *fp, const struct instruction *inst)
       break;
   }
 
-  if ((opcode == DDS_OP_ADR || opcode == DDS_OP_JEQ4) && inst->data.opcode.code & DDS_OP_FLAG_EXT)
-    vec[len++] = " | DDS_OP_FLAG_EXT";
-
-  type = DDS_OP_TYPE(inst->data.opcode.code);
-  assert(opcode == DDS_OP_PLM || type);
-  switch (type) {
-    case DDS_OP_VAL_1BY: vec[len++] = " | DDS_OP_TYPE_1BY"; break;
-    case DDS_OP_VAL_2BY: vec[len++] = " | DDS_OP_TYPE_2BY"; break;
-    case DDS_OP_VAL_4BY: vec[len++] = " | DDS_OP_TYPE_4BY"; break;
-    case DDS_OP_VAL_8BY: vec[len++] = " | DDS_OP_TYPE_8BY"; break;
-    case DDS_OP_VAL_STR: vec[len++] = " | DDS_OP_TYPE_STR"; break;
-    case DDS_OP_VAL_BST: vec[len++] = " | DDS_OP_TYPE_BST"; break;
-    case DDS_OP_VAL_SEQ: vec[len++] = " | DDS_OP_TYPE_SEQ"; break;
-    case DDS_OP_VAL_ARR: vec[len++] = " | DDS_OP_TYPE_ARR"; break;
-    case DDS_OP_VAL_UNI: vec[len++] = " | DDS_OP_TYPE_UNI"; break;
-    case DDS_OP_VAL_STU: vec[len++] = " | DDS_OP_TYPE_STU"; break;
-    case DDS_OP_VAL_ENU: vec[len++] = " | DDS_OP_TYPE_ENU"; break;
-    case DDS_OP_VAL_EXT: vec[len++] = " | DDS_OP_TYPE_EXT"; break;
+  if (opcode == DDS_OP_ADR || opcode == DDS_OP_JEQ4) {
+    if (inst->data.opcode.code & DDS_OP_FLAG_EXT)
+      vec[len++] = " | DDS_OP_FLAG_EXT";
+    switch (DDS_OP_TYPE(inst->data.opcode.code)) {
```

Comment:
I may be mistaken here, but shouldn't this be done for `DDS_OP_JEQ` as well?

If am *not* mistaken, `DDS_OP_JEQ` is never generated by IDLC and merely exists for backwards compatibility. Assuming that's indeed the case, then I think it would be best to eliminate (or `assert(0)`) the `DDS_OP_JEQ` cases in IDLC.

Reply:
You're right, `DDS_OP_JEQ` is not generated and still exists for backwards compatibility with topic descriptors compiled by earlier versions of IDLC. I've removed the code for JEQ from descriptor.c and added a remark in dds_opcodes.h that the instruction is deprecated.